src/api/app/models/staging/staging_project.rb

```diff
@@ -0,0 +1,144 @@
+# The staging project is the place where we collect a selection of packages and test, build and review them.
+module Staging
+  class StagingProject < Project
+    has_many :staged_requests, class_name: 'BsRequest', foreign_key: :staging_project_id, dependent: :nullify
+    has_many :status_reports, through: :repositories
+    belongs_to :staging_workflow, inverse_of: :staging_projects
+
+    def staging_identifier
+      name[/.*:Staging:(.*)/, 1]
+    end
+
+    def untracked_requests
+      requests_to_review - staged_requests
+    end
+
+    # The difference between staged requests and requests to review is that staged requests are assigned to the staging project.
+    # The requests to review are requests which are not related to the staging project (unless they are also staged).
+    # They simply need a review from the maintainers of the staging project.
+    def requests_to_review
+      @requests_to_review ||= BsRequest.with_open_reviews_for(by_project: name)
+    end
+
+    def building_repositories
+      set_buildinfo unless @building_repositories
+      @building_repositories
+    end
+
+    def broken_packages
+      set_buildinfo unless @broken_packages
+      @broken_packages
+    end
+
+    def missing_reviews
+      return @missing_reviews if @missing_reviews
+
+      @missing_reviews = []
+      attribs = [:by_group, :by_user, :by_project, :by_package]
+
+      (requests_to_review + staged_requests).uniq.each do |request|
+        request.reviews.each do |review|
+          next if review.state.to_s == 'accepted' || review.by_project == name
+          # FIXME: this loop (and the inner if) would not be needed
+          # if every review only has one valid by_xxx.
+          # I'm keeping it to mimic the python implementation.
```

Comment:
there is already Review.assigned_reviewer which does a very similiar loop

Reply:
Oh, right. But there are a few differences. I would keep the code as is and do that in a later refactoring. We are also going to drop the `requests_to_review` as discussed yesterday.